### PR TITLE
Fractional delay (windowed sinc)

### PR DESCRIPTION
**Pull request by f-brinkmann**
### Changes proposed in this pull request:

- Add new file `dsp_interpolation.py` because `dsp.py` is getting crowded.
- Add `pyfar.dsp.fractional_delay_sinc` function using the windowed sinc method
- Move `InterpolateSpectrum` to the new file
- - Add testing for new function

### CODE REVIEW

**Review by sikersten — APPROVED** [review]
Thanks for implementing! Only minor comments on docstring...

File: pyfar/dsp/dsp_interpolation.py [hunk 1]
```diff
@@ -0,0 +1,514 @@
...
+    delays to an input signal. A Kaiser window according to [#]_ Equations
+    (10.12) and (10.13) is used, which offers the possibility to control the
+    side lobe suppression.
+
+    Parameters
+    ----------
+    signal : Signal
+        The input data
+    delay : float, array like
+        The fractional delay (positive or negative). If this is a float, the
```
[the remark this answers is not in the document]
Comment (f-brinkmann) [reply pyfar/dsp/dsp_interpolation.py:24]: yes - good point

File: pyfar/dsp/dsp_interpolation.py [hunk 2]
```diff
@@ -0,0 +1,514 @@
...
+        same delay is applied to all channels of `signal`. If this is an array
+        like different delays are applied to the channels of `signal`. In this
+        case it must broadcast to `signal` (see `Numpy broadcasting
```
[the remark this answers is not in the document]
Comment (f-brinkmann) [reply pyfar/dsp/dsp_interpolation.py:27]: correct, changed.

File: pyfar/dsp/dsp_interpolation.py [hunk 3]
```diff
@@ -0,0 +1,514 @@
...
+
+        ``"cut"``
+            The delayed signal has the same length as the input signal but
+            parts of the signal that are shifted to values smaller than 0
+            samples and larger than ``signal.n_samples`` are removed from the
+            output
+        ``"cyclic"``
+            The delayed signal has the same length as the input signal. Parts
+            of the signal that are shifted to values smaller than 0 are wrapped
+            around the end. Parts that are shifted to values larger than
```
[the remark this answers is not in the document]
Comment (f-brinkmann) [reply pyfar/dsp/dsp_interpolation.py:48]: corrected

**Comment by mberz** [comment]
Please use a separate PR for restructuring code in the future. This way inter-dependencies of PRs can be easily avoided and code is easier to review ;)

**Pull request merged**